src/components/v5/common/CompletedAction/partials/PaymentBuilderEdit/hooks.ts

```diff
@@ -0,0 +1,156 @@
+import { unformatNumeral } from 'cleave-zen';
+import { BigNumber } from 'ethers';
+import moveDecimal from 'move-decimal-point';
+import { useMemo } from 'react';
+import { array, number, object, string } from 'yup';
+
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import useTokenLockStates from '~hooks/useTokenLockStates.ts';
+import { notNull } from '~utils/arrays/index.ts';
+import getLastIndexFromPath from '~utils/getLastIndexFromPath.ts';
+import { formatText } from '~utils/intl.ts';
+import { shouldPreventPaymentsWithTokenInColony } from '~utils/tokens.ts';
+import { amountGreaterThanZeroValidation } from '~utils/validation/amountGreaterThanZeroValidation.ts';
+import { CLAIM_DELAY_MAX_VALUE } from '~v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/partials/ClaimDelayField/consts.ts';
+import { allTokensAmountValidation } from '~v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/utils.ts';
+
+export const useValidationSchema = (networkInverseFee: string | undefined) => {
+  const { colony } = useColonyContext();
+  const colonyTokens = useMemo(
+    () =>
+      colony.tokens?.items
+        .filter(notNull)
+        .map((colonyToken) => colonyToken.token) || [],
+    [colony.tokens?.items],
+  );
+  const tokenLockStatesMap = useTokenLockStates();
+
+  return useMemo(
+    () =>
+      object()
+        .shape({
+          payments: array()
```

Comment:
This is almost the same as validation in `src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/hooks.ts`. Is there a way to reuse it?